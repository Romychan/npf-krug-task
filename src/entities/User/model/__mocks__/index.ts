import { LIMIT_USERS } from '../constants';

export const MOCK_USER_STORE = {
	limit: LIMIT_USERS,
	total: 0,
	isLoading: false,
	error: null,
	message: null,
	users: [],
};
