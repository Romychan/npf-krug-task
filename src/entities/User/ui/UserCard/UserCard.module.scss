@import '~/src/app/assets/styles/typography';

.profile {
	display: flex;
	gap: 1rem;
	align-items: center;
	width: 100%;
}

.information {
	overflow: hidden;
	display: flex;
	flex-direction: column;
}

.name,
.username {
	overflow: hidden;
	text-overflow: ellipsis;
}

.name {
	@include text-sm;

	color: var(--gray-900);
}

.username {
	@include text-xs;

	color: var(--gray-600);
}
