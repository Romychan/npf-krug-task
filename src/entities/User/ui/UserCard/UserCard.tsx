import { Avatar } from '~/shared/ui/Avatar';

import styles from './UserCard.module.scss';

interface IUserCardProps {
	/** The user's name */
	name: string;
	/** The user's username */
	username: string;
	/** The user's avatar */
	avatar?: string;
}

/** This component is used to display the user's avatar, name and username */
export const UserCard = ({ name, username, avatar = '' }: IUserCardProps) => {
	return (
		<div className={styles.profile} data-testid='user-card'>
			<Avatar src={avatar} name={username} alt={username} size='xs' isBordered />

			<div className={styles.information}>
				<span className={styles.name}>{name}</span>
				<span className={styles.username}>@{username}</span>
			</div>
		</div>
	);
};
