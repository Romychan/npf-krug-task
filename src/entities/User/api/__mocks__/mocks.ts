import { User } from '../../model/types';

export const MOCK_USER: User = {
	id: 1,
	name: 'First Last Name',
	email: 'test@test.test',
	username: 'username',
	image: 'https://robohash.org/L9Z.png?set=set4',
};
