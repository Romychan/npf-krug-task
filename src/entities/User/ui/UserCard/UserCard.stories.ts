import type { Meta, StoryObj } from '@storybook/react';

import { MOCK_USER } from '../../api/__mocks__/mocks';

import { UserCard } from './UserCard';

const meta = {
	title: 'Entities/UserCard',
	component: UserCard,
	parameters: {
		layout: 'centered',
	},
	tags: ['autodocs'],
} satisfies Meta<typeof UserCard>;

export default meta;
type Story = StoryObj<typeof meta>;

/** This state is used to display the default */
export const Default: Story = {
	args: {
		avatar: MOCK_USER.image,
		name: MOCK_USER.name,
		username: MOCK_USER.username,
	},
};

/** If the `avatar` is missing, the first three characters will be displayed */
export const WithoutAvatar: Story = {
	args: {
		name: MOCK_USER.name,
		username: MOCK_USER.username,
	},
};
