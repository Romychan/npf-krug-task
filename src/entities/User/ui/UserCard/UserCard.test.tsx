import { render, RenderResult, screen } from '@testing-library/react';

import { MOCK_USER } from '../../api/__mocks__/mocks';

import { UserCard } from './UserCard';

describe('UserCard', () => {
	let container: RenderResult;

	beforeEach(() => {
		container = render(<UserCard avatar={MOCK_USER.image} name={MOCK_USER.name} username={MOCK_USER.username} />);
	});

	it('should render correctly', () => {
		const cardElement = screen.getByTestId('user-card');

		expect(cardElement).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		expect(container).toMatchSnapshot();
	});
});
