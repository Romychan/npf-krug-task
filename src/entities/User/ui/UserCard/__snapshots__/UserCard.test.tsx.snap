// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserCard > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_profile_09a2f2"
        data-testid="user-card"
      >
        <div
          class="_container_15af9b _xs_15af9b _auto_15af9b _border_15af9b"
          data-testid="avatar"
        >
          <img
            alt="username"
            class="_image_15af9b"
            src="https://robohash.org/L9Z.png?set=set4"
          />
        </div>
        <div
          class="_information_09a2f2"
        >
          <span
            class="_name_09a2f2"
          >
            First Last Name
          </span>
          <span
            class="_username_09a2f2"
          >
            @
            username
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_profile_09a2f2"
      data-testid="user-card"
    >
      <div
        class="_container_15af9b _xs_15af9b _auto_15af9b _border_15af9b"
        data-testid="avatar"
      >
        <img
          alt="username"
          class="_image_15af9b"
          src="https://robohash.org/L9Z.png?set=set4"
        />
      </div>
      <div
        class="_information_09a2f2"
      >
        <span
          class="_name_09a2f2"
        >
          First Last Name
        </span>
        <span
          class="_username_09a2f2"
        >
          @
          username
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
